### PR TITLE
Jlc/add course run view set wrapper

### PR DESCRIPTION
**Pull request by johanv26**
## Description

Eox-nelp version for CourseRunViewSet in CMS
 feat: wrapper cms CourseRunViewSet view
The eox-nelp version endpoint CourseRunViewSet  is the same as CMS native version.

`{studio_url}/eox-nelp/api/v1/course_runs/` ihas the same result of  `{studio_url}/api/v1/course_runs/`
### Before
Only eox-info endpoint in cms eox-nelp urls.
### After

Peek 2023-10-26 15-14

### Related jira story

https://edunext.atlassian.net/browse/FUTUREX-616?atlOrigin=eyJpIjoiNTBkODkzOTM0ZTcxNDExMTgwY2ExNTVlMTVlMzhjZTUiLCJwIjoiaiJ9
## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/settings/common.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def plugin_settings(settings):
     settings.EOX_NELP_EDXMAKO_BACKEND = 'eox_nelp.edxapp_wrapper.backends.edxmako_m_v1'
     settings.EOX_NELP_BRANDING_BACKEND = 'eox_nelp.edxapp_wrapper.backends.branding_m_v1'
     settings.EOX_NELP_CERTIFICATES_BACKEND = 'eox_nelp.edxapp_wrapper.backends.certificates_m_v1'
+    settings.EOX_NELP_CMS_API_VIEWS = 'eox_nelp.edxapp_wrapper.backends.cms_api_views_m_v1'
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/settings/common.py:48]: YOu are right I passed it.

File: eox_nelp/edxapp_wrapper/test_backends/__init__.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from rest_framework.generics import ListAPIView, RetrieveAPIView
 from rest_framework.serializers import Serializer
 from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/test_backends/__init__.py]: SOlved in 2614e14608d26c435592796e0bda302f9ba50511

File: eox_nelp/settings/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/settings/common.py]: 5eb37be84590afdef704353c92d7d27af42c2273

File: eox_nelp/edxapp_wrapper/test_backends/cms_api_views_m_v1.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+class TestCourseRunViewSet(DummyViewset):
+    """Test version of course Run viewset"""
+
+    def list(self, request, *args, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
+        return Response(COURSE_RUN_TEST_RESPONSE)
+
+
+COURSE_RUN_TEST_RESPONSE = {
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/test_backends/cms_api_views_m_v1.py]: a2d3a16

File: eox_nelp/edxapp_wrapper/backends/cms_api_views_m_v1.py
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/backends/cms_api_views_m_v1.py:1]: 9592908

File: eox_nelp/cms/api/v1/views.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+"""
+CMS API v1 Views
+"""
+from eox_nelp.edxapp_wrapper.cms_api_views import CourseRunViewSet
+
+
+class NelpCourseRunViewSet(CourseRunViewSet):
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/cms/api/v1/views.py:7]: [43230d0](https://github.com/eduNEXT/eox-nelp/pull/103/commits/43230d0f3f78bb50d5cbfc68d8d8958b42489563)

File: eox_nelp/edxapp_wrapper/test_backends/cms_api_views_m_v1.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    Returns:
+        TestCourseRunViewSet class.
+    """
+    return TestCourseRunViewSet
+
+
+class TestCourseRunViewSet(DummyViewset):
+    """Test version of course Run viewset"""
+
+    def list(self, request, *args, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
```
[the remark this answers is not in the document]
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/test_backends/cms_api_views_m_v1.py]: I dont know it, I bring it from here.
https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/cms/djangoapps/api/v1/views/course_runs.py#L46
Is something like from Edx Lint.
So in this case I could removed.
[43230d0](https://github.com/eduNEXT/eox-nelp/pull/103/commits/43230d0f3f78bb50d5cbfc68d8d8958b42489563)

File: eox_nelp/edxapp_wrapper/cms_api.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+"""Wrapper cms api views.
+
+This contains  the required api view from cms/api/../views
+
+Attributes:
+    backend: Imported CMS api module by using the plugin settings.
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/cms_api.py:6]:
```suggestion
    cms_api: Imported CMS api module by using the plugin settings.
```

File: eox_nelp/edxapp_wrapper/cms_api.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+"""Wrapper cms api views.
+
+This contains  the required api view from cms/api/../views
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/cms_api.py:3]: This could contain all the api content not just views

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**